tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/CiamIntegrationTests.cs

```diff
@@ -213,6 +213,31 @@ public async Task OBOCiam_CustomDomain_ReturnsValidTokens()
             Assert.AreEqual(TokenSource.Cache, resultObo.AuthenticationResultMetadata.TokenSource);
         }
 
+        [TestMethod]
+        public async Task WithOidcAuthority_ValidatesIssuerSuccessfully()
+        {
+            //Get lab details
+            var labResponse = await LabUserHelper.GetLabUserDataAsync(new UserQuery()
+            {
+                FederationProvider = FederationProvider.CIAMCUD,
+                SignInAudience = SignInAudience.AzureAdMyOrg
+            }).ConfigureAwait(false);
+
+            //Test with different CIAM authority formats
+            string[] authorities =
+            {
+                string.Format("https://{0}.ciamlogin.com/", labResponse.User.LabName),
+                string.Format("https://{0}.ciamlogin.com/{1}.onmicrosoft.com", labResponse.User.LabName, labResponse.User.LabName),
+                string.Format("https://{0}.ciamlogin.com/{1}", labResponse.User.LabName, labResponse.Lab.TenantId),
+                string.Format("https://login.msidlabsciam.com/{0}/v2.0/", labResponse.Lab.TenantId)
+            };
+
+            foreach (var authority in authorities)
+            {
+                await RunCiamCCATest(authority, labResponse.App.AppId).ConfigureAwait(false);
```

Comment:
This will mostly use the `WithAuthority` API in MSAL, which gets MSAL to parse the authority and to understand it's a CIAM authority, in which case it won't perform OIDC discovery. 

IMO, you want to force test the `WithOidcAuthority` route, which performs OIDC discovery. I think you want to test it with AAD and with CIAM authorities.

Reply:
In the latest commit that `RunCiamCCATest()` now takes in an extra boolean parameter so the tests themselves control which authority API to use.